.github/workflows/codeql.yml

```diff
@@ -10,26 +10,29 @@ jobs:
    CodeQL-Build:
       # CodeQL runs on ubuntu-latest and windows-latest
       runs-on: ubuntu-latest
+      permissions:
+         contents: read
+         security-events: write
 
       steps:
          - name: Checkout repository
-           uses: actions/checkout@v3
```

Comment:
Are we supposed to ping to sha instead of versions ex: https://github.com/Azure/Bridge-To-Kubernetes/blob/main/.github/workflows/codeql-analysis.yml#L49

Reply:
we're moving over to semantic versioning/pinning major versions since that's what the community overwhelmingly asked for (even though it's not technically best practice)

Reply:
I agree with @hsubramanianaks. Our workflows should use sha, especially when pointing towards actions not owned by us.

What we agreed on wasn't to use the tag in our workflows, just to expose the major version tag when we release our workflows (but recommend using the sha in the action docs). This matches what github owned actions do. That is not related to this.

Reply:
yes good points i've update with hashes